src/odemis/gui/model/__init__.py

```diff
@@ -661,28 +661,43 @@ def add_new_feature(self, pos_x, pos_y, pos_z=None, f_name=None, milling_angle=D
             pos_z = self.main.focus.position.value['z']
         feature = CryoFeature(f_name, pos_x, pos_y, pos_z, milling_angle)
         self.main.features.value.append(feature)
+        self.main.currentFeature.value = feature
         return feature
 
     # Todo: find the right margin
     ATOL_FEATURE_POS = 0.1e-3  # m
 
     def select_current_position_feature(self):
         """
-        Given current stage position, either select one of the features closest to the position or create a new one with the position
+        Given current stage position, either select one of the features closest to
+          the position or create a new one with the position.
         """
         current_position = self.main.stage.position.value
-        for feature in self.main.features.value:
-            feature_dist = math.hypot(feature.pos.value[0] - current_position["x"],
-                                      feature.pos.value[1] - current_position["y"])
-            if feature_dist <= self.ATOL_FEATURE_POS:
-                self.main.currentFeature.value = feature
-                break
-        else:
-            # create new feature if no close feature found
-            feature = self.add_new_feature(current_position["x"], current_position["y"],
-                                                          self.main.focus.position.value["z"])
-            logging.debug("A new feature is created at {} because none are close by.".format((current_position["x"], current_position["y"])))
-            self.main.currentFeature.value = feature
+        current_feature = self.main.currentFeature.value
+
+        def dist_to_pos(feature):
+            return math.hypot(feature.pos.value[0] - current_position["x"],
+                              feature.pos.value[1] - current_position["y"])
+
+
+        if current_feature and dist_to_pos(current_feature) <= self.ATOL_FEATURE_POS:
+            return  # We are already good nothing else to do
+
+        # Find the closet feature (which is nearby)
+        try:
+            closest = min(self.main.features.value, key=dist_to_pos)
+            if dist_to_pos(closest) <= self.ATOL_FEATURE_POS:
+                self.main.currentFeature.value = closest
+                return
+        except ValueError:  # raise by min() when no features
+            pass
```

Comment:
You pass here but I would expect the rest of the code in this except part. If it doesn't go into this expect it returns so that is actually clearer to follow.

Reply:
The problem is that the rest of the code is both for the "except" part, and for the "else" of the if. That's why it's after.